cea/demand/airconditioning_model.py

```diff
@@ -27,248 +28,304 @@
-    temp_zone_set : zone air set point temperature (°C), [=Ta from calc thermal loads, including control strategy]
-    qv_req : ventilation requirements at time step t according to the mixed occupancy schedule of the building (m3/s)
-    qe_sen : sensible heating or cooling load to be supplied by the HVAC system at time step t in [W]
-    temp_5_prev : zone air temperature at time step t-1 (°C)
-    wint : internal moisture gains at time step t according to the mixed occupancy schedule of the building (kg/s)
-    gv : object of class globalvar
-    timestep : hour of the year [0..8760]
+    """
+    Calculate AC air mass flows, energy demand and temperatures for the cooling case
 
-    Returns
-    -------
-    qe_hs_sen
-    qe_cs_sen
-    qe_hum
-    qe_dehum
-    pel_hum_aux
-    ma_hs
-    ma_cs
-    ts_hs
-    ts_cs
-    tr_hs
-    tr_cs
-    w2
-    w3
-    t5
+    :param tsd: time series data dict
+    :param hoy: time step
+    :param gv: globalvars
+    :return:
     """
 
+    temp_zone_set = tsd['theta_a'][hoy]  # zone set temperature according to scheduled set points
+    qe_sen = tsd['Qcs_sen'][hoy]  # get the total sensible load from the RC model, in this case without any mechanical ventilation losses??? NO!
+    m_ve_hvac_req = tsd['m_ve_mech'][hoy] # mechanical ventilation flow rate according to ventilation control
+    wint = tsd['w_int'][hoy] # internal moisture gains from occupancy
+
+    # sensible cooling load provided by conditioning of ventilation air
+    #  = enthalpy difference in air mass flow between supply and room temperature
+    q_cs_sen_ventilation = calc_hvac_sensible_cooling_ventilation_air(tsd, hoy, gv)
+
+    # calculate the part of the sensible load that is supplied by conditioning the recirculation air
+    # = additional load that has to be covered by conditioning room air through recirculation
+    # additional load can not be smaller than 0, over cooling situation
+    q_cs_sen_recirculation = min(qe_sen - q_cs_sen_ventilation, 0) / 1000  # (kW)
+
+    # indoor air set point
+    t5 = temp_zone_set
+
+    # ventilation air properties after heat exchanger
+    t2, w2 = calc_hex(tsd, hoy)  # (°C), (kg/kg)
+
+    # State No. 3
+    # Assuming that AHU does not modify the air humidity
+    w3v = w2  # virtual moisture content at state 3
+
+    # determine virtual state in the zone without moisture conditioning
+    # moisture balance accounting for internal moisture load, see also Eq. (4.32) in [1]
+    w5_prime = (wint + w3v * m_ve_hvac_req) / m_ve_hvac_req
+
+    # supply air condition
+    t3 = gv.temp_sup_cool_hvac
+
+    # room supply moisture content:
+    # algorithm for cooling case
+    w3 = calc_w3_cooling_case(t5, w2, t3, w5_prime)
+
+    # State of Supply
+    ts = t3  # minus expected delta T rise in the ducts TODO: check and document value of temp decrease
+    ws = w3
+
+    # actual room condition
+    w5 = (wint + w3 * m_ve_hvac_req) / m_ve_hvac_req
```

Comment:
@JIMENOFONSECA Oops. That should not happen. I haven't encountered that. Do you remember which scenario? I usually test with the Open scenario and the Zug scenario.